src/components/NavBar/ChainSelector.tsx

```diff
@@ -72,6 +70,12 @@ export const ChainSelector = ({ leftAlign }: ChainSelectorProps) => {
 
   const theme = useTheme()
 
+  const [chains, setChains] = useState(NETWORK_SELECTOR_CHAINS)
+  const showTestnets = useAtomValue(showTestnetsAtom)
+  useEffect(() => {
+    setChains(NETWORK_SELECTOR_CHAINS.filter((chain) => !(chain === SupportedChainId.SEPOLIA && !showTestnets)))
+  }, [showTestnets])
```

Comment:
skip the filter step if `showTestnets` is true?

```suggestion
  useEffect(() => {
    if (!showTestnets) {
          setChains(NETWORK_SELECTOR_CHAINS.filter((chain) => !(chain === SupportedChainId.SEPOLIA)))
    }
  }, [showTestnets])
```

Reply:
also this may be a needless optimization for a small array, but we don't really need a filter step at all. we could have 2 arrays (prodnets and testnets) and just do something like this:

`const chains = showTestnets ? NETWORK_SELECTOR_CHAINS : [... PRODUCTION_CHAINS , ...TESTNET_CHAINS]`